### PR TITLE
docs: add migration guide for v5.0.0

### PR DESCRIPTION
**Pull request by snitin315**
- [ ] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### CODE REVIEW

File: migration-v5.md [hunk 1]
```diff
@@ -0,0 +1,316 @@
+# Migration guide
+
+This document serves as a migration guide for `webpack-dev-server@5.0.0`.
+
+## ⚠ BREAKING CHANGES
+
+- Minimum supported `Node.js` version is `14.15.0`.
```
Comment (alexander-akait) [inline migration-v5.md:7]: let's do it `16` :smile: I think it is time to drop 14 node.js again :smile:

**Review by alexander-akait — APPROVED** [review]
Good job, thank you

File: migration-v5.md (hunk 1 above)
Comment (snitin315) [reply migration-v5.md:7]:
```suggestion
- Minimum supported `Node.js` version is `16.10.0`.
```

**Comment by alexander-akait** [comment]
@snitin315 Feel free to merge when it will be ready

**Pull request merged**